tests/src/Interop/IJW/NativeVarargs/IjwNativeVarargs.cpp

```diff
@@ -0,0 +1,474 @@
+            values[12],
+            values[13],
+            values[14],
+            values[15],
+            values[16],
+            values[17],
+            values[18],
+            values[19],
+            values[20],
+            values[21],
+            values[22],
+            values[23],
+            values[24],
+            values[25],
+            values[26],
+            values[27],
+            values[28],
+            values[29],
+            values[30],
+            values[31],
+            values[32],
+            values[33],
+            values[34],
+            values[35],
+            values[36],
+            values[37],
+            values[38],
+            values[39],
+            values[40]
+        );
+        bool result = expected == actual;
+        if (!result)
+        {
+            std::cout << "RunHFAsTest Failed:" << "Expected:" << expected << '\t' << "Actual:" << actual << std::endl;
+        }
+        return result;
+    }
+
+    static float AggregateHFAs(float current, HFA hfa)
+    {
+        return current + hfa.f1 + hfa.f2 + hfa.f3 + hfa.f4;
+    }
+
+    bool RunLongLongsTest(System::Random^ rng)
+    {
+        std::array<long long, NumArgsPerCall> values;
+        for (std::size_t i = 0; i < NumArgsPerCall; ++i)
+        {
+            values[i] = rng->Next(System::Int32::MinValue, System::Int32::MaxValue);
+        }
+
+        auto expected = std::accumulate(values.begin(), values.end(), 0LL, std::plus<>{});
+        auto actual = SumLongLongs(NumArgsPerCall,
+            values[0],
+            values[1],
+            values[2],
+            values[3],
+            values[4],
+            values[5],
+            values[6],
+            values[7],
+            values[8],
+            values[9],
+            values[10],
+            values[11],
+            values[12],
+            values[13],
+            values[14],
+            values[15],
+            values[16],
+            values[17],
+            values[18],
+            values[19],
+            values[20],
+            values[21],
+            values[22],
+            values[23],
+            values[24],
+            values[25],
+            values[26],
+            values[27],
+            values[28],
+            values[29],
+            values[30],
+            values[31],
+            values[32],
+            values[33],
+            values[34],
+            values[35],
+            values[36],
+            values[37],
+            values[38],
+            values[39],
+            values[40]
+        );
+        bool result = expected == actual;
+        if (!result)
+        {
+            std::cout << "RunLongLongsTest Failed:" << "Expected:" << expected << '\t' << "Actual:" << actual << std::endl;
+        }
+        return result;
+    }
+
+    bool RunPromotedShortsTest(System::Random^ rng)
+    {
+        std::array<int, NumArgsPerCall / 2> intValues;
+        std::array<short, NumArgsPerCall - (NumArgsPerCall / 2)> shortValues;
```

Comment:
Unfortunately these types will probably not be widened. I do not believe we should expect this to work. I believe we support passing unwidened types, to be read as unwidened types for backwards compatibility.

Reply:
As we spoke offline, C++/CLI emits IL to do the widening for us.